src/python/WMCore/MicroService/Unified/MSRuleCleaner.py

```diff
@@ -3,22 +3,142 @@
+    """
+    def __init__(self, message):
+        self.myMessage = "MSRuleCleanerArchival: %s" % message
+        super(MSRuleCleanerArchival, self).__init__(self.myMessage)
+
+
+class MSRuleCleanerCleanup(MSRuleCleanerException):
+    """
+    Cleanup Exception Class for MSRuleCleaner Module in WMCore MicroServices
+    """
+    def __init__(self, message):
+        self.myMessage = "MSRuleCleanerCleanup: %s" % message
+        super(MSRuleCleanerCleanup, self).__init__(self.myMessage)
+
+
+class MSRuleCleanerWflow(dict):
+    """
+    A minimal workflow and transfer information representation to serve the needs
+    of the MSRuleCleaner Micro Service.
+    """
+
+    def __init__(self, doc, **kwargs):
+        super(MSRuleCleanerWflow, self).__init__(**kwargs)
+
+        # Search for all the keys we need from the ReqManager workflow describtion
+        myDoc = {}
+        for tup in self.docSchema():
+            if tup[0] in doc:
+                myDoc[tup[0]] = deepcopy(doc[tup[0]])
+            else:
+                myDoc.update({tup[0]: tup[1]})
+        self.update(myDoc)
+
+    def docSchema(self):
+        """
+        Return the data schema for the document.
+        It's a tuple where:
+        * 1st element: is the key name / attribute in the request
+        * 2nd element: is the default value
+        * 3rd element: is the expected data type
+
+        Template format:
+            {
+            "RequestName": "ReqName",
+            "RequestType": "Type",
+            "RequestStatus": "Status",
+            "OutputDatasets": ["list of output datasets"],
+            'RulesToClean': {'plineMSTrCont': [],
+                             'plineMSTrBlock': [],
+                             'plineAgentCont': [],
+                             'plineAgentBlock': []},
+            'CleanupStatus': {'plineMSTrCont': False,
+                              'plineMSTrBlock': False,
+                              'plineAgentCont': False,
+                              'plineAgentBlock': False},
+            "TransferInfo": {}           # A dictionary like object representing the transfer
+                                         # information - returned by the MSOutput REST call.
+            'TargetStatus': 'normal-archived' || 'rejected-achived' || 'aborted-archived',
+            'ParentageResolved': Bool,
+            'PlineMarkers': [None],
+            'IsClean': False
+            }
+        :return: a list of tuples
+        """
+        docTemplate = [
+            ('RequestName', None, (str, unicode)),
+            ('RequestType', None, (str, unicode)),
+            ('RequestStatus', None, (str, unicode)),
+            ('OutputDatasets', [], list),
+            ('RulesToClean', {'plineMSTrCont': [],
+                              'plineMSTrBlock': [],
+                              'plineAgentCont': [],
+                              'plineAgentBlock': []}, dict),
+            ('CleanupStatus', {'plineMSTrCont': False,
```

Comment:
As we discussed it in a private call, we will be keeping the number of pipe lines for the moment, and I am now  following a different path when it comes to the creation of those keys here, so that  we avoid errors like typos etc.